index.d.ts

```diff
@@ -0,0 +1,40 @@
+import * as Promise from 'bluebird';
+import TypedError = require('typed-error');
+
+declare module 'bluebird-lru-cache' {
+
+	export class NoSuchKeyError extends TypedError {
+		public constructor(
+			public key: string,
+		) {
+			super();
+		}
+	}
+
+	export interface Options {
+		max: number;
+		maxAge: number;
+		length: (n: any) => number;
```

Comment:
You'll want to mostly match the types in https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lru-cache/index.d.ts and use generics to get rid of all these `any`s in the same way